mteb/evaluation/evaluators/RerankingEvaluator.py

```diff
@@ -188,6 +186,19 @@ def compute_metrics_individual(self, model):
 
         return {"map": mean_ap, "mrr": mean_mrr}
 
+    def _encode_unqiue_texts(self, all_texts, encode_queries_func):
```

Comment:
Thank you for the useful changes! Adding the `_encode_unqiue_texts `method to improve encoding efficiency is a great idea. I have a few minor suggestion:
Correct the typo in the name from `_encode_unqiue_texts` to `_encode_unique_texts`.